azurerm/resource_arm_network_interface.go

```diff
@@ -497,13 +514,21 @@ func flattenNetworkInterfaceIPConfigurations(ipConfigs *[]network.InterfaceIPCon
 		props := ipConfig.InterfaceIPConfigurationPropertiesFormat
 
 		niIPConfig["name"] = *ipConfig.Name
-		niIPConfig["subnet_id"] = *props.Subnet.ID
+
+		if props.Subnet != nil {
+			niIPConfig["subnet_id"] = *props.Subnet.ID
+		}
+
 		niIPConfig["private_ip_address_allocation"] = strings.ToLower(string(props.PrivateIPAllocationMethod))
 
 		if props.PrivateIPAllocationMethod == network.Static {
 			niIPConfig["private_ip_address"] = *props.PrivateIPAddress
 		}
 
+		if props.PrivateIPAddressVersion != "" {
+			niIPConfig["private_ip_address_version"] = props.PrivateIPAddressVersion
```

Comment:
can we cast this enum to a string?

```suggestion
			niIPConfig["private_ip_address_version"] = string(props.PrivateIPAddressVersion)
```